MedicalInstitution.js

```diff
@@ -0,0 +1,197 @@
+const fs = require("fs");
+
+function getDateTimeFormatted(date = new Date()) {
+  const day = date.getDate().toString().padStart(2, "0");
+  const month = (date.getMonth() + 1).toString().padStart(2, "0");
+  const year = date.getFullYear().toString().padStart(4, "0");
+  const hours = date.getHours().toString().padStart(2, "0");
+  const minutes = date.getMinutes().toString().padStart(2, "0");
+
+  return `[${day}.${month}.${year} ${hours}:${minutes}]`;
+}
+
+function log(message) {
+  console.log(`${getDateTimeFormatted()} ${message}`);
+  fs.appendFile(
+    "medicalInstitutionLog.txt",
+    `${getDateTimeFormatted()} ${message} \n`,
+    (err) => {
+      if (err) {
+        console.error("Error writing to the file:", err);
+      } else {
+        console.log("Data written to the file successfully!");
+      }
+    }
+  );
+}
+
+class Person {
+  constructor(name, lastName) {
+    this.name = name;
+    this.lastName = lastName;
+  }
+}
+
+class Doctor extends Person {
+  pacientList = [];
+  appointmentList = [];
+  constructor(name, lastName, spec) {
+    super(name, lastName);
+    this.spec = spec;
+  }
+
+  setPacient(pacient) {
+    if (pacient instanceof Person) {
+      this.pacientList = [...this.pacientList, pacient];
+      log(
+        `${this.name} has accepted ${pacient.name} ${pacient.lastName} as his pacient`
+      );
+    } else {
+      console.log(pacient, "is not a person");
+    }
+  }
+
+  makeAppointment(pacient, appointmentDate, type) {
+    if (this.pacientList.includes(pacient)) {
+      if (type == "pressure") {
+        let newAppointment = new BloodPressureTest(
+          this,
+          pacient,
+          appointmentDate
+        );
+        this.appointmentList.push(newAppointment);
+        log(
+          `Dr.${this.name} made a appointment for a blood pressure test for pacient ${pacient.name}`
+        );
+        return newAppointment;
+      }
+      if (type == "sugar") {
+        let newAppointment = new BloodSugarTest(this, pacient, appointmentDate);
+        this.appointmentList.push(newAppointment);
+        log(
+          `Dr.${this.name} made a appointment for a blood blood sugar for pacient ${pacient.name}`
+        );
+        return newAppointment;
+      }
+      if (type == "cholesterol") {
+        let newAppointment = new BloodSugarTest(this, pacient, appointmentDate);
+        this.appointmentList.push(newAppointment);
+        log(
+          `Dr.${this.name} made a appointment for a blood cholesterol test for pacient ${pacient.name}`
+        );
+        return newAppointment;
+      } else {
+        console.log("Test of type", type, "does not exist");
+      }
```

Comment:
Because we have return in the first if, we do not have to have the else below

```suggestion
      if (type == "cholesterol") {
        let newAppointment = new BloodSugarTest(this, pacient, appointmentDate);
        this.appointmentList.push(newAppointment);
        log(
          `Dr.${this.name} made a appointment for a blood cholesterol test for pacient ${pacient.name}`
        );
        return newAppointment;
      }
 console.log("Test of type", type, "does not exist");
```